t/09_qesapdeployment.t

```diff
@@ -406,4 +406,74 @@ subtest '[qesap_ansible_script_output] root' => sub {
 };
 
+subtest '[qesap_create_aws_credentials]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @calls;
+    my @contents;
+
+    $qesap->redefine(script_output => sub { return '/path/to/aws/credentials/file'; });
+    $qesap->redefine(assert_script_run => sub { push @calls, $_[0]; });
+    $qesap->redefine(save_tmp_file => sub { push @contents, @_; });
+    $qesap->redefine(autoinst_url => sub { return 'http://10.0.2.2/tests/'; });
+    $qesap->redefine(qesap_get_file_paths => sub {
+            my %paths;
+            $paths{qesap_conf_trgt} = '/BRUCE';
+            return (%paths);
+    });
+
+    qesap_create_aws_credentials('MY_KEY', 'MY_SECRET');
+
+    note("\n  C-->  " . join("\n  C-->  ", @calls));
+    note("\n  C-->  " . join("\n  C-->  ", @contents));
+    ok((any { qr|mkdir -p ~/\.aws| } @calls), '.aws directory initialized');
+    ok((any { qr|curl.+/files/credentials.+/path/to/aws/credentials/file| } @calls), 'AWS Credentials file downloaded');
+    ok((any { qr|cp /path/to/aws/credentials/file ~/\.aws/credentials| } @calls), 'AWS Credentials copied to ~/.aws/credentials');
+    is $contents[0], 'credentials', "AWS credentials file: credentials is the expected value and got $contents[0]";
+    like $contents[1], qr/aws_access_key_id = MY_KEY/, "Expected key MY_KEY is in the credentials file";
+    like $contents[1], qr/aws_secret_access_key = MY_SECRET/, "Expected secret MY_SECRET is in the credentials file";
+};
+
+subtest '[qesap_create_aws_config]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @calls;
+    my @contents;
+
+    $qesap->redefine(script_output => sub { return 'eu-central-1'; });
+    $qesap->redefine(assert_script_run => sub { push @calls, $_[0]; });
+    $qesap->redefine(save_tmp_file => sub { push @contents, @_; });
+    $qesap->redefine(autoinst_url => sub { return 'http://10.0.2.2/tests/'; });
+    $qesap->redefine(qesap_get_file_paths => sub {
+            my %paths;
+            $paths{qesap_conf_trgt} = '/BRUCE';
+            return (%paths);
+    });
+
+    qesap_create_aws_config();
+
+    note("\n  C-->  " . join("\n  C-->  ", @calls));
+    note("\n  C-->  " . join("\n  C-->  ", @contents));
+    ok((any { qr|mkdir -p ~/\.aws| } @calls), '.aws directory initialized');
+    ok((any { qr|curl.+/files/config.+~/\.aws/config| } @calls), 'AWS Config file downloaded');
+    ok((any { qr/eu-central-1/ } @calls), 'AWS Region matches');
+    is $contents[0], 'config', "AWS config file: config is the expected value and got $contents[0]";
+    like $contents[1], qr/region = eu-central-1/, "Expected region eu-central-1 is in the config file";
+};
+
+subtest '[qesap_remote_hana_public_ips]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @calls;
+
+    $qesap->redefine(get_required_var => sub { return 'EC2'; });
```

Comment:
you do not need to do it, just call `set_var` (at least it is the approach used in other UT in this repo)

Take care to also set to undef before the end of the test 
```
set_var('YOUR_VAR', undef);
```

Reply:
Will change it.